src/lightning_utilities/core/imports.py

```diff
@@ -84,67 +84,48 @@ def compare_version(package: str, op: Callable, version: str, use_base_version:
 
 
 class RequirementCache:
-    """Boolean-like class for check of requirement with extras and version specifiers.
+    """Boolean-like class to check for requirement and module availability.
+
+    Args:
+        requirement: The requirement to check, version specifiers are allowed.
+        module: The optional module to try to import if the requirement check fails.
 
     >>> RequirementCache("torch>=0.1")
     Requirement 'torch>=0.1' met
     >>> bool(RequirementCache("torch>=0.1"))
     True
     >>> bool(RequirementCache("torch>100.0"))
     False
-    """
-
-    def __init__(self, requirement: str) -> None:
-        self.requirement = requirement
-
-    def _check_requirement(self) -> None:
-        if not hasattr(self, "available"):
-            try:
-                pkg_resources.require(self.requirement)
-                self.available = True
-                self.message = f"Requirement {self.requirement!r} met"
-            except Exception as ex:
-                self.available = False
-                self.message = f"{ex.__class__.__name__}: {ex}. HINT: Try running `pip install -U {self.requirement!r}`"
-
-    def __bool__(self) -> bool:
-        """Format as bool."""
-        self._check_requirement()
-        return self.available
-
-    def __str__(self) -> str:
-        """Format as string."""
-        self._check_requirement()
-        return self.message
-
-    def __repr__(self) -> str:
-        """Format as string."""
-        return self.__str__()
-
-
-class ModuleAvailableCache:
```

Comment:
This reverts #86 and removes ModuleAvailableCache? But we are already using it, so shouldn't we keep it?

Reply:
I'm fine with keeping it. I'm not sure about what's the expected "deprecation policy" of this package.

Reply:
lets keep it as it would break some past versions